aiida/orm/nodes/node.py

```diff
@@ -63,34 +71,34 @@ def _warn_if_not_entered(self, method):
 
             warnings.warn(msg, AiidaDeprecationWarning)  # pylint: disable=no-member
 
-    def __enter__(self):
+    def __enter__(self) -> Union[IO[str], IO[bytes]]:
         self._was_entered = True
         return self._fileobj.__enter__()
 
-    def __exit__(self, *args):
+    def __exit__(self, *args: Any) -> None:
         self._fileobj.__exit__(*args)
 
-    def __getattr__(self, key):
+    def __getattr__(self, key: str):
```

Comment:
not necessarily, right?
what about `_fileobj`?

Reply:
not sure what you mean here, `__getattr__` always receives a string

Reply:
ah, my bad, I was thinking about the return value.
I got confused by the fact that it doesn't need to be specified for the __ methods